Description: enable testcases build and run.
 One testcase can't build because of missing up to date semver-java:
 - TestVersionSemantics
 Three testcases fail on every arch while the library seems functional:
 - TestUri99LaunchOnReservedCharPathBug908
 - TestStructGen01
 - TestStructGen02.
Author: Gilles Filippini <pini@debian.org>
--- gluegen2.orig/make/build-test.xml
+++ gluegen2/make/build-test.xml
@@ -84,6 +84,7 @@
             <pathelement location="${junit.jar}" />
             <pathelement location="${semver.jar}" />
             <pathelement location="${gluegen.jar}" />
+            <pathelement location="/usr/share/java/jsr305.jar" />
         </path>
 
         <path id="junit.run.classpath">
--- gluegen2.orig/make/scripts/runtest.sh
+++ gluegen2/make/scripts/runtest.sh
@@ -42,7 +42,7 @@ rm -f $LOG
 GLUEGEN_ROOT=`dirname $builddir`
 ROOTREL_BUILD=`basename $builddir`
 
-X_ARGS="-Drootrel.build=$ROOTREL_BUILD -Dgluegen.root=$GLUEGEN_ROOT"
+X_ARGS="-Drootrel.build=$ROOTREL_BUILD -Dgluegen.root=$GLUEGEN_ROOT -Djava.library.path=$builddir/obj:$builddir/test/build/natives"
 #D_ARGS="-Djogamp.debug.ProcAddressHelper -Djogamp.debug.NativeLibrary -Djogamp.debug.NativeLibrary.Lookup"
 #D_ARGS="-Djogamp.debug.TraceLock"
 #D_ARGS="-Djogamp.debug.Platform -Djogamp.debug.NativeLibrary"
@@ -64,12 +64,12 @@ X_ARGS="-Drootrel.build=$ROOTREL_BUILD -
 #D_ARGS="-Djogamp.debug.IOUtil -Djogamp.debug.IOUtil.Exe"
 #D_ARGS="-Djogamp.debug.ByteBufferInputStream"
 #D_ARGS="-Djogamp.debug.Bitstream"
-#D_ARGS="-Djogamp.debug=all"
+D_ARGS="-Djogamp.debug=all"
 #D_ARGS="-Djogamp.debug.Logging"
 
 function onetest() {
     #USE_CLASSPATH=lib/junit.jar:$ANT_JARS:lib/semantic-versioning/semver.jar:"$builddir"/../make/lib/TestJarsInJar.jar:"$builddir"/gluegen-rt.jar:"$builddir"/gluegen.jar:"$builddir"/gluegen-test-util.jar:"$builddir"/test/build/gluegen-test.jar
-    USE_CLASSPATH=lib/junit.jar:$ANT_JARS:lib/semantic-versioning/semver.jar:"$builddir"/../make/lib/TestJarsInJar.jar:"$builddir"/gluegen-rt.jar:"$builddir"/gluegen.jar:"$builddir"/gluegen-test-util.jar:"$builddir"/test/build/gluegen-test.jar:"$builddir"/gluegen-rt-natives.jar
+    USE_CLASSPATH=/usr/share/java/junit4.jar:$ANT_JARS:lib/semantic-versioning/semver.jar:"$builddir"/../test/TestJarsInJar/TestJarsInJar.jar:"$builddir"/gluegen-rt.jar:"$builddir"/gluegen.jar:"$builddir"/gluegen-test-util.jar:"$builddir"/test/build/gluegen-test.jar:"$builddir"/gluegen-rt-natives.jar
     #USE_CLASSPATH=lib/junit.jar:$ANT_JARS:lib/semantic-versioning/semver.jar:"$builddir"/../make/lib/TestJarsInJar.jar:"$builddir"/gluegen-rt-alt.jar:"$builddir"/gluegen.jar:"$builddir"/gluegen-test-util.jar:"$builddir"/test/build/gluegen-test.jar
     libspath="$builddir"/test/build/natives
     #USE_CLASSPATH=lib/junit.jar:$ANT_JARS:"$builddir"/../make/lib/TestJarsInJar.jar:"$builddir"/classes:"$builddir"/test/build/classes
@@ -88,66 +88,68 @@ function onetest() {
     echo
 }
 #
+(cd $builddir/../test/TestJarsInJar && sh make.sh)
+#
-#onetest com.jogamp.common.GlueGenVersion 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestSystemPropsAndEnvs 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestVersionInfo 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestVersionNumber 2>&1 | tee -a $LOG
+onetest com.jogamp.common.GlueGenVersion 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestSystemPropsAndEnvs 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestVersionInfo 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestVersionNumber 2>&1 | tee -a $LOG
 #onetest com.jogamp.common.util.TestVersionSemantics 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestIteratorIndexCORE 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.locks.TestRecursiveLock01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.locks.TestRecursiveThreadGroupLock01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.locks.TestSingletonServerSocket00 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.locks.TestSingletonServerSocket01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.locks.TestSingletonServerSocket02 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestFloatStack01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestIntegerStack01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestArrayHashSet01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestArrayHashMap01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.IntIntHashMapTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.IntObjectHashMapTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.LongIntHashMapTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestPlatform01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestRunnableTask01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestIteratorIndexCORE 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.locks.TestRecursiveLock01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.locks.TestRecursiveThreadGroupLock01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.locks.TestSingletonServerSocket00 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.locks.TestSingletonServerSocket01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.locks.TestSingletonServerSocket02 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestFloatStack01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestIntegerStack01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestArrayHashSet01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestArrayHashMap01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.IntIntHashMapTest 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.IntObjectHashMapTest 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.LongIntHashMapTest 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestPlatform01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestRunnableTask01 2>&1 | tee -a $LOG
 onetest com.jogamp.common.util.TestIOUtil01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestTempJarCache 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestJarUtil 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestValueConversion 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestSyncRingBuffer01 $*
-#onetest com.jogamp.common.util.TestLFRingBuffer01 $*
-#onetest com.jogamp.common.util.TestBitfield00 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestBitstream00 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestBitstream01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestBitstream02 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestBitstream03 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.util.TestBitstream04 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.TestUrisWithAssetHandler 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.TestUriQueryProps 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.TestUri01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.TestUri02Composing 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.TestUri03Resolving 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.TestUri99LaunchOnReservedCharPathBug908 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.AssetURLConnectionUnregisteredTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.net.AssetURLConnectionRegisteredTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.junit.sec.TestSecIOUtil01 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.BuffersTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.TestBuffersFloatDoubleConversion 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.TestPointerBufferEndian 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.TestStructAccessorEndian 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.TestByteBufferInputStream 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.TestByteBufferOutputStream 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.nio.TestByteBufferCopyStream 2>&1 | tee -a $LOG
-#onetest com.jogamp.common.os.TestElfReader01 $* 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.test.junit.internals.TestType 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestTempJarCache 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestJarUtil 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestValueConversion 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestSyncRingBuffer01 $*
+onetest com.jogamp.common.util.TestLFRingBuffer01 $*
+onetest com.jogamp.common.util.TestBitfield00 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestBitstream00 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestBitstream01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestBitstream02 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestBitstream03 2>&1 | tee -a $LOG
+onetest com.jogamp.common.util.TestBitstream04 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.TestUrisWithAssetHandler 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.TestUriQueryProps 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.TestUri01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.TestUri02Composing 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.TestUri03Resolving 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.TestUri99LaunchOnReservedCharPathBug908 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.AssetURLConnectionUnregisteredTest 2>&1 | tee -a $LOG
+onetest com.jogamp.common.net.AssetURLConnectionRegisteredTest 2>&1 | tee -a $LOG
+onetest com.jogamp.junit.sec.TestSecIOUtil01 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.BuffersTest 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.TestBuffersFloatDoubleConversion 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.TestPointerBufferEndian 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.TestStructAccessorEndian 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.TestByteBufferInputStream 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.TestByteBufferOutputStream 2>&1 | tee -a $LOG
+onetest com.jogamp.common.nio.TestByteBufferCopyStream 2>&1 | tee -a $LOG
+onetest com.jogamp.common.os.TestElfReader01 $* 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.internals.TestType 2>&1 | tee -a $LOG
 
-#onetest com.jogamp.gluegen.test.junit.generation.PCPPTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.jcpp.IncludeAbsoluteTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.jcpp.CppReaderTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.jcpp.TokenPastingWhitespaceTest 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.jcpp.PreprocessorTest 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.generation.PCPPTest 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.jcpp.IncludeAbsoluteTest 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.jcpp.CppReaderTest 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.jcpp.TokenPastingWhitespaceTest 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.jcpp.PreprocessorTest 2>&1 | tee -a $LOG
 
-#onetest com.jogamp.gluegen.test.junit.generation.Test1p1JavaEmitter 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.test.junit.generation.Test1p2ProcAddressEmitter 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.test.junit.generation.Test1p2LoadJNIAndImplLib 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.test.junit.structgen.TestStructGen01 2>&1 | tee -a $LOG
-#onetest com.jogamp.gluegen.test.junit.structgen.TestStructGen02 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.generation.Test1p1JavaEmitter 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.generation.Test1p2ProcAddressEmitter 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.generation.Test1p2LoadJNIAndImplLib 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.structgen.TestStructGen01 2>&1 | tee -a $LOG
+onetest com.jogamp.gluegen.test.junit.structgen.TestStructGen02 2>&1 | tee -a $LOG
 
